core/zip/inc/Compression.h

```diff
@@ -39,56 +39,71 @@ namespace ROOT {
-   kUncompressedLevel = 0,
-   // Compression level reserved when we are not sure what to use (1 is for the fastest compression)
-   kUseMinCompressionLevel = 1,
-   kDefaultZLIB = 1,
-   kDefaultLZ4 = 4,
-   kDefaultOld = 6,
-   kDefaultLZMA = 7
+struct RCompressionSetting {
+   enum class EDefaults {
+      /// Use the global compression setting for this process; may be affected by rootrc.
+      kUseGlobal = 0,
+      /// Use the compile-time default setting
+      kUseCompiledDefault = 404,
+      /// Use the default analysis setting; fast reading but poor compression ratio
+      kUseAnalysis = 404,
+      /// Use the recommended general-purpose setting; moderate read / write speed and compression ratio
+      kUseGeneralPurpose = 101,
+      /// Use the setting that results in the smallest files; very slow read and write
+      kUseSmallest = 207
+   };
+   enum class ELevel {
+      /// Some objects use this value to denote that the compression algorithm
+      /// should be inherited from the parent object
+      kInherit = -1,
+      // Compression level reserved for "uncompressed state"
+      kUncompressed = 0,
+      // Compression level reserved when we are not sure what to use (1 is for the fastest compression)
+      kUseMin = 1,
+      kDefaultZLIB = 1,
+      kDefaultLZ4 = 4,
+      kDefaultOld = 6,
+      kDefaultLZMA = 7
+   };
+   enum class EAlgorithm {
+      /// Some objects use this value to denote that the compression algorithm
+      /// should be inherited from the parent object (e.g., TBranch should get the algorithm from the TTree)
+      kInherit = -1,
+      /// Use the global compression algorithm
+      kUseGlobal = 0,
+      /// Use ZLIB compression
+      kZLIB,
+      /// Use LZMA compression
+      kLZMA,
+      /// Use the old compression algorithm
+      kOldCompressionAlgo,
+      /// Use LZ4 compression
+      kLZ4,
+      /// Undefined compression algorithm (must be kept the last of the list in case a new algorithm is added).
+      kUndefined
+   };
 };
 
-enum ECompressionSetting {
-   /// Use the global compression setting for this process; may be affected by rootrc.
-   kUseGlobalCompressionSetting = 0,
-   /// Use the compile-time default setting
-   kUseCompiledDefaultCompressionSetting = 404,
-   /// Use the default analysis setting; fast reading but poor compression ratio
-   kUseAnalysisCompressionSetting = 404,
-   /// Use the recommended general-purpose setting; moderate read / write speed and compression ratio
-   kUseGeneralPurposeCompressionSetting = 101,
-   /// Use the setting that results in the smallest files; very slow read and write
-   kUseSmallestCompressionSetting = 207
+enum ECompressionAlgorithm {
+   // Deprecated name, do *not* use:
+   kInherit = static_cast<int>(RCompressionSetting::EAlgorithm::kInherit),
+   /// Deprecated name, do *not* use:
+   kUseGlobalSetting = static_cast<int>(RCompressionSetting::EAlgorithm::kUseGlobal),
```

Comment:
these used to be called `kInheritCompressionAlgorithm` and `kUseGlobalCompressionAlgorithm`, I think code that was using `ECompressionAlgorithm::kInheritCompressionAlgorithm` won't compile...we have no test that is using those?

Reply:
@bluehood Note that the diff is misleading.  Those are 'new' names introduced by a recent patch and never release, we are attempting to make those new names 'better' before the release.

Reply:
@pcanal @oshadura `kUseGlobalCompressionSetting` seems to be released in v6.14 (see [here](https://github.com/root-project/root/blob/v6-14-00-patches/core/zip/inc/Compression.h#L45)).

Also wouldn't this PR change the integer values of `ECompressionAlgorithm` with respect to v6.14?

Reply:
kUseGlobalCompressionSetting is in process of deprecation https://github.com/root-project/root/blob/4aa3e31999ff3569ce0a3e88b887102e590961bc/core/zip/inc/Compression.h#L49

Reply:
No, int numbers is never been changed.

Reply:
yes, I see. Here you right it should be `kUseGlobalCompressionSetting` and not `kUseGlobalSetting`

Reply:
Actually from tests I realized we need both of them :)